Adapting to Linux

grandchamp opened the pull request.

grandchamp committed:
Adapting to Linux

matgr1 commented:
First of all, thanks for the PR!

It looks like there won't be a much better solution to this kind of thing until dotnet/coreclr/issues/930 is resolved...

However, I'd like to put this in a separate branch if you don't mind since this is a lot of duplicated code and hopefully there will be a better way someday...  I've created a branch called Linux that you can use

Also, could you could switch it around to be "IsLinux" instead of "IsWindows"? It currently works as is for Windows and macOS, so it seems that Linux needs the special case here.

matgr1 commented:
a couple of other things:

can you stick with the free image namespace (as in "FreeImageAPI" instead of "Library")?

and change "FreeImageStaticImportsWindows.cs" to "FreeImageStaticImportsDefault.cs"

and one more quick test if you don't mind: would a symlink work as an alternative workaround until the dotnet core issue is resolved?

grandchamp committed:
Changing StaticImportsWindows to Default, since it works for Windows and OSX;
Changing IsWindows property to IsLinux, since Linux is the special case;
Moving projectUrl to packOptions;

grandchamp commented:
Ok, done!

Yes, symlink works fine:
`sudo ln -s /usr/lib/x86_64-linux-gnu/libfreeimage.so /usr/lib/FreeImage`
